### PR TITLE
DSD-1050: Header timeout functions

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1050](https://jira.nypl.org/browse/DSD-1050)

## This PR does the following:

- Reimplements the code to completely log out a patron from Encore/Catalog in the Header.
- The code is based on https://github.com/NYPL/dgx-header-component/blob/master/src/utils/encoreCatalogLogOutTimer.js

The logic flow is as follows:
- The `Header` component mounts
- The patron is not logged in:
  - Cool, delete the `nyplIdentityPatron` and `VALID_DOMAIN_LAST_VISITED` cookies if they exist, just to make sure we log the patron out completely.
- The patron is logged in:
  - The patron is on a "valid" domain or the `VALID_DOMAIN_LAST_VISITED` is not available
    - Create the `VALID_DOMAIN_LAST_VISITED` with the current time and log the patron out in 30 minutes.
  -  The patron is not on a "valid" domain
      - Get the time they were last seen on a valid domain through the `VALID_DOMAIN_LAST_VISITED` cookie.
        - They were not last seen on a valid domain, so immediately log them out.
        - Use the last time they were seen on a valid domain to get a new session timeout. Log the patron out in that new time.

When the patron is logged out, the `PAT_LOGGED_IN`, `VALID_DOMAIN_LAST_VISITED`, and `nyplIdentityPatron` cookies are deleted.

## How has this been tested?

Locally and waiting ⏲️  for the function to log me out from Encore and the Catalog while being on `local.nypl.org`.

## Accessibility concerns or updates

- The loaded iframe is set to `display: none` so it should not be a problem.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
@nonword if you get a chance, please review this. The Vercel preview site won't work to test this since the cookies are not available on that domain.

**Review by nonword — APPROVED** [review]
The core logic looks essentially identical to the original to me

File: src/components/Header/utils/encoreCatalogLogOutTimer.ts
Comment (bigfishdesign13) [inline src/components/Header/utils/encoreCatalogLogOutTimer.ts:84]: The comment above refers to setting a value to `2 seconds`, but here is it set to 1 second.  Why the descrepancy?

**Review by bigfishdesign13 — APPROVED** [review]
This looks good, but I did not really test it.  I am marking as `approved`, but let me know if you'd like to see more thorough testing before merging.

File: src/components/Header/utils/encoreCatalogLogOutTimer.ts
Comment (EdwinGuzman) [reply src/components/Header/utils/encoreCatalogLogOutTimer.ts:84]: True, I updated it because I didn't think the test should have that much of a time out. I'll update the comment

File: src/components/Header/Header.tsx
Comment (oliviawongnyc) [inline src/components/Header/Header.tsx:74]: Was it always 30 minutes? I have a recollection of it being an hour, but could be wrong!

File: src/components/Header/utils/encoreCatalogLogOutTimer.ts
Comment (oliviawongnyc) [inline src/components/Header/utils/encoreCatalogLogOutTimer.ts]: Nit: Typo in word, "patron".

**Review by oliviawongnyc — COMMENTED** [review]
I read this over several times and think I understand it, but definitely found it hard to visualize. Might it be useful to include a section on this logic in the Header storybook page?

File: src/components/Header/Header.tsx
Comment (EdwinGuzman) [reply src/components/Header/Header.tsx:74]: For this timer, it always was 30 minutes. Maybe there's something else that's an hour?

**Comment by EdwinGuzman** [comment]
> Might it be useful to include a section on this logic in the Header storybook page? 

Yes. That page is actually empty. I'll create a ticket that will be just to fill out that page, how to use it, its logic, etc. But, I'll ask you to fill out a few sections like the cookie login logic and the focus-trap implementation in the drop downs.

**Pull request merged**
